### PR TITLE
Update team meetings and deliverables workflow

### LINKED ISSUE #201: Document a team process around sprint planning

### PR DESCRIPTION
**Pull request by choldgraf**
This PR updates our team workflows after a recent team meeting to discuss our Deliverables review meetings and how it fits in with board workflows. It also updates guidelines about meetings in general that we have recently discussed.

Major updates:

- Deliverables meeting is now weekly, with a noted plan to move back to bi-weekly
- Remove the dev/org backlogs from "official" team process, but includes description of them in case it's useful
- Simplifies language around the activity board and deliverables structure
- Updates our meeting templates with this new structure
- Encodes official meeting roles in the templates+docs

closes #201

### CODE REVIEW

File: practices/meetings.md
Comment (sgibson91) [inline practices/meetings.md:18]: Do we also want to add a notetaker? I find it super hard to moderate and take notes simultaneously for example.

**Review by sgibson91 — APPROVED** [review]
LGTM! I have one tiny Q but isn't a big blocker.

File: practices/meetings.md
Comment (choldgraf) [reply practices/meetings.md:18]: Great point - will add that in before merge

Comment (yuvipanda) [reply practices/meetings.md:18]: I'd also suggest adding a timekeeper!

Comment (sgibson91) [reply practices/meetings.md:18]: This is a good reference for these roles https://cfe.unc.edu/facilitator-recorder-and-timekeeper-roles/

**Comment by choldgraf** [comment]
hey @2i2c-org/tech-team - I've updated this PR and expanded its scope a bit to encompass our team meeting discussions. It now tries to encode the changes that we suggested in that meeting. LMK what you think!

File: practices/development.md
Comment (damianavila) [inline practices/development.md:14]: If we are having the deliverable meeting on Tue, then the cycle should start on Tue, right?

File: practices/development.md
Comment (damianavila) [inline practices/development.md:16]:
```suggestion
At the end of each week, we should have completed all of the major deliverables set out for that week.
```

Comment (damianavila) [inline practices/development.md:16]: > At the end of each week

Should we push that to include the next Monday? Or do we keep that as buffer time?

File: practices/development.md
Comment (damianavila) [inline practices/development.md:40]: Again a reference to the end of the week...
This is actually pushing me to think that we are duplicating some stuff with the team sync issue and the deliverables meeting... I mean, the reason to put the deliverable meeting on Tue was having some time on Mon to read through the team sync issue. But the team sync issue has a back / record piece and a next (todo) piece that should be completed after the deliverable meeting. So there is a conflict there...
One way to solve it is just to **remove** the todo part from the team sync issue and trigger that one on Fri morning. Then, we can move the deliverable meetings to Mon and all things align, thoughts?

File: practices/development.md
Comment (damianavila) [inline practices/development.md:82]: So, you are proposing to have per-project boards, I think that is a nice idea as long as we do not have dozen of boards :wink:
I guess the "current" deliverable board is actually one encompassing project that eventually would be partitioned into smaller ones. And I guess the org board could be seen as one of those particular projects, right?

File: practices/meetings.md
Comment (damianavila) [inline practices/meetings.md:56]:
```suggestion
- Keep track of the current time, and provide the Facilitator / meeting attendees cues that help them decide when to move things forward.
```

File: .github/ISSUE_TEMPLATE/meeting-deliverables.md
Comment (yuvipanda) [inline .github/ISSUE_TEMPLATE/meeting-deliverables.md:8]: Let's just call it sprint planning meeting? I want to try align with pre-existing terminology as much as possible.

File: .github/ISSUE_TEMPLATE/meeting-deliverables.md
Comment (yuvipanda) [inline .github/ISSUE_TEMPLATE/meeting-deliverables.md:24]: I'd like for us to make one board per week, so we can actually go through all of them from the last week. Helps make explicit decisions on what to continue working on, and what to drop / deprioritize.

File: .github/ISSUE_TEMPLATE/meeting-deliverables.md
Comment (yuvipanda) [inline .github/ISSUE_TEMPLATE/meeting-deliverables.md:28]: I like it!

File: .github/ISSUE_TEMPLATE/meeting-deliverables.md
Comment (yuvipanda) [inline .github/ISSUE_TEMPLATE/meeting-deliverables.md:30]: I REALLY LIKE THIS!

File: .github/ISSUE_TEMPLATE/meeting-team.md
Comment (yuvipanda) [inline .github/ISSUE_TEMPLATE/meeting-team.md:30]: Nice!

File: practices/development.md
Comment (yuvipanda) [inline practices/development.md:25]: These should be per-week boards that we archive at the end

File: practices/development.md
Comment (yuvipanda) [inline practices/development.md:51]: If we move to per-week boards, the board should have:

1. Ready to work
2. In progress
3. Done

`Up Next` should be the backlog, and live elsewhere.

File: practices/development.md
Comment (yuvipanda) [inline practices/development.md:59]: In the PR description or comment, you should also mention what kind of review you want. We should also use draft PRs effectively - any team member can provide code review on PRs marked as not-a-draft, but only requested team members need to look at PRs marked as drafts. This lets us both ask for specific review early, but provide opportunity for everyone to review things before merging.

File: practices/development.md
Comment (yuvipanda) [inline practices/development.md:60]: I think it should always be particular team members, to avoid diffusion of responsibility. Marking a PR as not-a-draft should help solicit reviews from others.

File: practices/development.md
Comment (yuvipanda) [inline practices/development.md:79]: How about instead we just have *one* board, where the goal is purely prioritization? It has just high level tasks, and only contains an ordered list of deliverables that can be picked up. Nothing is automatically put into this board - someone must develop an issue to a point where it is ready to be worked on, and explicitly puts it in this board with appropriate priority ordering. We can then use this to pick up tasks from when needed.

**Review by yuvipanda — CHANGES_REQUESTED** [review]
Thanks a lot for writing this up, @choldgraf!

File: .github/ISSUE_TEMPLATE/meeting-deliverables.md
Comment (damianavila) [reply .github/ISSUE_TEMPLATE/meeting-deliverables.md:24]: In general, I agree.
See my comment about that here: https://github.com/2i2c-org/team-compass/pull/193#discussion_r682934433

File: practices/development.md
Comment (damianavila) [reply practices/development.md:59]: > but provide opportunity for everyone to review things before merging.

Yep, I think it is important to give enough time (to be defined) so everyone can look at "ready" PRs before merging them (unless they are already deployed as we usually do).

File: practices/development.md
Comment (damianavila) [reply practices/development.md:60]: > I think it should always be particular team members, to avoid diffusion of responsibility.

👍 

> Marking a PR as not-a-draft should help solicit reviews from others.

Let's make that clear in the docs... if the PR is in "ready" state then, please, jump into reviewing it!

File: practices/development.md
Comment (choldgraf) [reply practices/development.md:40]: Good point that we're duplicating some information - I think moving our planning meeting to weekly does now duplicate some actions with the deliverables sync.

So you're proposing something like:

- Monday: Have sprint planning meeting, set deliverables to accomplish this week.
- Friday: Have a team sync issue where team members report on progress for their items. Focus this issue more around "what are the major things I did, where did I encounter challenges, and what are items I couldn't get to". Remove the "what will I do next" steps because we'll cover this in the next sprint planning meeting.
- The next Monday: we can focus that meeting around planning next steps rather than covering "what happened last week" because people have already filled out their issue.

Is that right?

File: practices/development.md
Comment (choldgraf) [reply practices/development.md:82]: Well - here I am not trying to set a team **policy**, but suggesting a pattern that we can use and replicate. That said, yes I am suggesting that we track ongoing bodies of work with boards, and use those boards to help prioritize major things to work on next within a given project. I think this is project-specific and those boards should be closed as they are completed. That said, some projects might prefer something like GitHub Milestones to boards, to me the important thing is that we have *some* method for prioritizing deliverables, keeping track of the "big picture" and planning things out.

File: .github/ISSUE_TEMPLATE/meeting-deliverables.md
Comment (choldgraf) [reply .github/ISSUE_TEMPLATE/meeting-deliverables.md:24]: Is the main benefit of "one board per week" so that we have a historical record of previous weeks? I can see benefit in that, though also worry that it is adding toil (somebody has to create the new board each week and close previous boards) as well as some complexity in our workflow since you'll have to remember a new URL each week for that week's board.

The alternative would be to do something like #191 to clean up the board at the end of each week.

I don't feel strongly about this though. If you really think "1 board per week" is the way to go, I'm fine trying that out.

File: practices/development.md
Comment (choldgraf) [reply practices/development.md:51]: That makes sense to me (though, we should refine our methods for tracking priorities within a given project, otherwise these meetings are going to involve a lot of "digging through issues" that may not be very efficient, I suspect)

quick question:

What do we do with deliverables that are important to create, but that require more discussion and brainstorming before somebody can just "do them"? An example is https://github.com/2i2c-org/team-compass/issues/175 - that issue still requires discussion, but it's an important discussion to have and does take effort from team members throughout the week. How would we track something like that? (I think in previous iterations this is what would have gone in a "Needs refinement" column).

One option is that we follow the same workflow, but use a github label like `needs:decision`, and the person "assigned" to the issue is in charge of moving forward the discussion and converging on a decision.

File: practices/development.md
Comment (choldgraf) [reply practices/development.md:59]: Let's tackle this one in #175 and we can update the language here as needed if the PR hasn't been merged yet?

File: practices/development.md
Comment (choldgraf) [reply practices/development.md:60]: I think this is a good policy as well - the action of going from "draft" to "not draft" should be meaningful for our team 👍 - can we tackle this in #175 ?

File: practices/development.md
Comment (choldgraf) [reply practices/development.md:79]: Mmm - I like the idea of keeping track of prioritizes and high-level deliverables. Though have a few concerns with "one board to rule them all".

1. It doesn't track the overall structure and plan of work in a given project, so will be less useful at understanding the "big picture" of *why* a particular deliverable should have priority. This will also make it harder to know which deliverables go at the top of the list.
2. Many of our projects are totally different from one another, and some team members will have workstreams that don't overlap (e.g. myself and @damianavila are doing a lot of stuff on Jupyter Book projects...would it feel weird to have those deliverables interwoven with things like `pilot-hubs/` upgrades? How would we prioritize one deliverable over another in these cases?
3. I think the "Someone must develop and issue" process is actually a **really important** process, and I'm a bit worried that we are being kind-of hand-wavy about how it happens. For complex issues, once we've gotten to the point of "somebody just needs to implement it", a lot of work has already been done. How do we use these workflows to ensure that this happens?

**Comment by choldgraf** [comment]
@yuvipanda I left a few responses to your comments throughout! In general I agree w/ them all, and certainly agree with the spirit of them. I have a few comments/suggestions/concerns about the specific implementations we're suggesting, and would love to know what others think!

File: .github/ISSUE_TEMPLATE/meeting-deliverables.md
Comment (yuvipanda) [reply .github/ISSUE_TEMPLATE/meeting-deliverables.md:24]: > Is the main benefit of "one board per week" so that we have a historical record of previous weeks?

The main benefit to me is the very tight scoping. There are 0 cards that can be ignored, there is a very clear reason each card is there. It also gives us a concrete way to look at cards that were not done the previous week, and make a clear decision on wether to carry it forward, descope, or what. It's ok to carry it forward (many tasks will take longer than a week!), but it becomes an explicit decision than something implicit.

The alternative to one board per week is to completely erase the board and start again. But one board per week is simpler (IMO). The process would be:

1. Before the meeting, create a new board
2. During the meeting, populate new board. A combination of carry over tasks from previous week and new cards
3. Archive current board in its state

File: practices/development.md
Comment (yuvipanda) [reply practices/development.md:51]: > What do we do with deliverables that are important to create, but that require more discussion and brainstorming before somebody can just "do them"?

File: practices/development.md
Comment (yuvipanda) [reply practices/development.md:79]: Agree that (3) is super important, and I responded to some of that in https://github.com/2i2c-org/team-compass/pull/188#discussion_r683591706. I think most work is iterative in different kinds of work, and it is difficult to get them to a point where "somebody just needs to implement it" happens. I think doing that kind of work should just be part of the sprint - maybe for one sprint half the work is just getting an issue fleshed out and decisions made (if necessary) with other people.

For (2), I think it's ok to have them mixed in for now. If absolutely necessary, we can filter by labels (or repos). Let's try it out and see if it is a problem?

For (1), IMO we should rely on the content of the issue, since most of it will be pretty unstructured. I don't know if  moving them across columns in a board provides more information than text in an issue. Since almost all our work interacts with folks from outside 2i2c, and they don't really expect to be using our boards, we should just continue using issues until we hit more specific problems there.

Anyway, I think (3) is the most important thing to do. I don't think adding 1 board per project will help with that though - you still have to remember and check which board to look at, and do the work. Keeping it in the issue, and not treating it differently from implementation work alone will help reduce diffusion of responsibility too

File: .github/ISSUE_TEMPLATE/meeting-deliverables.md
Comment (choldgraf) [reply .github/ISSUE_TEMPLATE/meeting-deliverables.md:24]: Yeah that makes sense - maybe I am just biasing things towards my own workflows? I like bookmarking things, so it is helpful for me to have "one URL to go to this week's activity board", and I don't have to keep re-updating the URL each week because it is now a new board. If others don't think this is inconvenient then I can live with it though.

File: practices/development.md
Comment (choldgraf) [reply practices/development.md:51]: yeah I think the challenge is that it is harder to scope cards that need more discussion etc, since it requires more input from others and such. That said, I agree with your proposal here so let's just give this a shot and see if it's awkward in some spaces that make us want to update

File: practices/development.md
Comment (choldgraf) [reply practices/development.md:79]: How do you imagine this being different from the "Development Backlog" that we have already been using? Is the main difference that it is more disciplined and refined than the current one? It feels like the current dev backlog wasn't working for folks, so I'm trying to understand where this differs to make sure we don't just create another board that folks don't find useful.

**Comment by choldgraf** [comment]
OK I've updated this PR with a few changes from the comments in this PR.

1. Renamed Deliverables Review to "Sprint Planning"
2. The meeting is now on Monday.
  - **I propose we have our first Sprint Meeting next Monday, August 9th, at 8:00AM Pacific**
3. Team Sync issues now happen on Fridays, and are more about retrospective rather than looking ahead (since we'll use the Sprint Planning meeting for this)
4. Sprint boards are now created and archived for each sprint.
5. I've re-purposed the "Development backlog" to be a "Deliverables backlog" and have added a description of how to use this board.

**Comment by choldgraf** [comment]
One other thought about the Deliverables board:

I think we should do a triage of the issues in the deliverables backlog (https://github.com/orgs/2i2c-org/projects/7) to make sure that it accurately reflects the work that we want to do. Ideally this is work that should be done **before** the Sprint Planning meeting, but I'm not sure the best way to incorporate this into our practices. A few options come to mind:

1. Designate a person to be in charge of the deliverables backlog...their job is to clean up the backlog and make sure that the items there are properly-scoped etc. They'd do this with a lot of one-on-one communication with people that have more context about each deliverable.
2. Incorporate a review of the backlog into the Sprint Meeting (but I suspect this will boost the meeting above 30 minutes)
3. Create a dedicated meeting to review and update the backlog (maybe with a subset of team members?)

**Comment by choldgraf** [comment]
Since this hasn't yet been merged, and we potentially have not yet converged, let's just stick with our pre-planned meeting time and have the normal team deliverables sync on Tuesday. I'll spend Monday trying to clean up our backlog a bit and we can run the meeting like a sprint planning meeting, even if we'll begin on a Monday the next time around.

File: practices/development.md
Comment (sgibson91) [inline practices/development.md:36]:
```suggestion
This may include coordinating team discussing and arriving at a decision before work can be done - in either case, the person assigned to the deliverable is responsible for carrying out whatever work is needed to close it. [^invest]
```

**Review by sgibson91 — APPROVED** [review]

File: practices/development.md
Comment (damianavila) [reply practices/development.md]: Yep, indeed, that was my proposal :wink:

File: practices/development.md
Comment (damianavila) [reply practices/development.md:107]: Sounds reasonable to me. Thanks for the additional clarification.

**Comment by damianavila** [comment]
> I think we should do a triage of the issues in the deliverables backlog (https://github.com/orgs/2i2c-org/projects/7) to make sure that it accurately reflects the work that we want to do. Ideally this is work that should be done before the Sprint Planning meeting, but I'm not sure the best way to incorporate this into our practices. 


IMHO, triaging the backlog should be done by the team instead of just one person.
I would totally invest 15-30 more minutes in the meeting to approach that collectively, IMHO.

File: .github/ISSUE_TEMPLATE/meeting-sprint-planning.md
Comment (damianavila) [inline .github/ISSUE_TEMPLATE/meeting-sprint-planning.md:10]:
```suggestion
This is a **30 minute** recurring meeting every **Monday at 8:00AM Pacific time** (here's [**a timezone converter**](https://arewemeetingyet.com/Los%20Angeles/2000-01-01/08:00/2i2c%20Team%20Meeting#eyJ1cmwiOiJodHRwczovL2hhY2ttZC5pby9ZNVNCTXhWN1I2Q01xemVUWGdtNWtBIn0=), ignore the date!).
```

**Comment by choldgraf** [comment]
Hey all - I just pushed another update with some more guidance about the breakdown of deliverables/issues/etc. I think that we are relatively converged on a pattern to try out here? Let me know if there's anything outstanding to change.

**Comment by choldgraf** [comment]
@damianavila I agree that the deliverables refinement should happen with participation from the team as well. Do you think that this is something that _all_ team members should be present for, or could we accomplish this via more focused meetings throughout the week?

E.g., I was imagining that we could have a role like "Backlog Steward" we could rotate on a 1-2 month basis, and that role's job would be to coordinate with team members in focused meetings to update deliverables on the backlog. Those meetings could be "public to any team member, but with only 1-2 people we are scheduling around".

(that said, I'd also be in favor of extending our weekly planning meeting to add a 30 minute deliverables backlog refinement session, just throwing this idea out there to see if it resonates)

**Review by GeorgianaElena — APPROVED** [review]
Extremely grateful for everyone's work and ideas towards improving our team practices. Also, thank you for trying this out and sorry for missing the first iterations of it.

I like the direction in which this is going and how the things seem more simple and have a more straightforward implementation path.

I left a couple of comments about the things that sounded a little confusing for me.

File: .github/ISSUE_TEMPLATE/meeting-sprint-planning.md
Comment (choldgraf) [reply .github/ISSUE_TEMPLATE/meeting-sprint-planning.md:23]: good point - will re-name to `team sync update` 👍

File: practices/development.md
Comment (choldgraf) [reply practices/development.md:41]: I think that the structure should work for enhancements or admin stuff, though I suspect that bugs might benefit from its own dedicated issue structure (e.g. since it's often important to note environment details etc).

Maybe we should re-name this from `new-enhancement` to `new-deliverable`? We could use labels to track what kind of deliverable it was (e.g. admin, development, etc)

File: practices/development.md
Comment (choldgraf) [reply practices/development.md:50]: good point - I'll rename here to "Benefit"

For "acceptance criteria" - good point, I will re-work!

**Comment by choldgraf** [comment]
thanks for these comments @GeorgianaElena ! check out the latest commit, which I think addresses them

**Comment by damianavila** [comment]
> @damianavila I agree that the deliverables refinement should happen with participation from the team as well. Do you think that this is something that all team members should be present for, or could we accomplish this via more focused meetings throughout the week?

That is a thought one!! I am not sure... I see benefits and cons to both approaches. Maybe we can pick one and try it and see how that works out for real?

File: practices/development.md
Comment (damianavila) [inline practices/development.md:55]:
```suggestion
  All deliverables should have either a set of concrete steps to take to meet the deliverable, or at least one task with the **acceptance criteria** for when the deliverable will be complete.
```

**Comment by choldgraf** [comment]
@damianavila how about we start off with some meetings with subsets of team members, and see how this feels from an "information silo" perspective. We can revisit in our team discussions if we think it'd be useful to have an "all hands" style meeting for this.

**Comment by damianavila** [comment]
> @damianavila how about we start off with some meetings with subsets of team members, and see how this feels from an "information silo" perspective. We can revisit in our team discussions if we think it'd be useful to have an "all hands" style meeting for this.

I am OK with that.

> This could also potentially be something that we work into our "one on one" practices. Review major work items that are in the backlog, clean them up and reprioritize as we get new context, etc.

I am not OK with that 😉  because that automatically puts you as the Backlog Steward and that should be a rotating thing (as you imply in the subset proposal you made?). Unless you are referring to "one on one" as a broader thing and not our weekly 1:1 with you.

**Comment by choldgraf** [comment]
@damianavila I agree with you! I meant "general team 1 on 1 practices". But, I agree with you that right now that would effectively just mean myself because I'm the one with the most 1-on-1s with everybody :-)

It feels to me like we should define this "Backlog Steward" role explicitly, and rotate through it similar to the Support Steward. The Backlog Steward's job would be to facilitate 1-2 weekly "backlog refinement meetings" with one or more other team members.

**Comment by choldgraf** [comment]
What's the pathway towards getting this merged, and iterating later? 

@yuvipanda you've requested changes, can you approve or make comments about further changes?
@damianavila do you feel that we should define the Backlog Steward role before merging this PR? Or something to track in ongoing iterations?

**Comment by damianavila** [comment]
> It feels to me like we should define this "Backlog Steward" role explicitly, and rotate through it similar to the Support Steward. The Backlog Steward's job would be to facilitate 1-2 weekly "backlog refinement meetings" with one or more other team members.

I would agree with that.

> @damianavila do you feel that we should define the Backlog Steward role before merging this PR? Or something to track in ongoing iterations?

I would be OK with ongoing iterations (modulo we capture those things in another ticket). Approving now.

**Review by damianavila — APPROVED** [review]

**Comment by choldgraf** [comment]
Sounds good - I've opened up https://github.com/2i2c-org/team-compass/issues/205 to track the creation of this role

**Comment by choldgraf** [comment]
OK all - we've had several rounds of iteration and reviews that I think are now addressed. I've got approvals from 3/4 folks and I think that @yuvipanda is happy with what remains. I'll plan to merge this tomorrow (Saturday) unless I hear otherwise, and our next sprint meeting will be on **Monday**.

**Pull request merged**
